### PR TITLE
feat: override strict for javascript module

### PR DESCRIPTION
**Pull request by colinaaa**
## Summary
This is essentially a port of https://github.com/webpack/webpack/pull/18555.

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/rspack-test-tools/tests/configCases/parsing/override-strict/rspack.config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/** @type {import("@rspack/core").Configuration[]} */
+module.exports = [
+	{
+		mode: "production",
+		entry: ["./strict"],
+		module: {
+			parser: {
+				'javascript/auto': {
```
Comment (colinaaa) [inline packages/rspack-test-tools/tests/configCases/parsing/override-strict/rspack.config.js:8]: This is a little different with webpack. Using `parser.javascript` does not work for this case: https://github.com/webpack/webpack/pull/18555/files#diff-7dbe21b12d5bd21e85c33e276954bcb7e6fc06371746dc76db783d77d8b4eb91R8

File: crates/rspack_core/src/options/module.rs [hunk 2]
```diff
@@ -193,6 +193,22 @@ impl ExportPresenceMode {
...
+  Strict,
+  NoneStrict,
+}
+
+impl From<&str> for OverrideStrict {
+  fn from(value: &str) -> Self {
+    match value {
+      "strict" => Self::Strict,
+      "non-strict" => Self::NoneStrict,
+      _ => unreachable!("parser.overrideStrict should be 'strict' or 'non-strict'"),
```
Comment (colinaaa) [inline crates/rspack_core/src/options/module.rs:207]: I'm not sure if it is a good choice to use `unreachable!`

**Comment by LingyuCoder** [comment]
LGTM, cc @ahabhgk

**Review by LingyuCoder — APPROVED** [review]

File: crates/rspack_core/src/options/module.rs (hunk 2 above)
Comment (ahabhgk) [reply crates/rspack_core/src/options/module.rs:207]: It's ok, invalid configuration will be rejected by zod

**Review by ahabhgk — APPROVED** [review]

**Pull request merged**